test/client.js

```diff
@@ -148,6 +153,24 @@ describe('client', function () {
     client.connect(url, user, pass, done);
   });
 
+  it('should send an error on ENOTFOUND', function (done) {
+    client.connect(
+        hostIsNotReachableUrls.ENOTFOUND, user, pass, function (err) {
+      if (err && err.name === 'HostIsNotReachable') {
+        done();
+      }
+    });
```

Comment:
your tests need an assert.fail("message") for the else case so that if the test fails, you get a descriptive message quickly instead of an ambiguous test timeout.